main/resolve/src/mill/resolve/Resolve.scala

```diff
@@ -148,7 +148,7 @@ object Resolve {
       allowRepeats = false,
       allowLeftover = ep.argSigs0.exists(_.reader.isLeftover)
     ).flatMap { (grouped: TokenGrouping[_]) =>
-      val mainData = ep.asInstanceOf[MainData[_, Any]]
+      val mainData = ep.asInstanceOf[MainData[Any, Any]]
```

Comment:
Not sure why this is necessary, something about type inference must have changed in 0.6.0 to make it source incompatible (even though it's binary compatible)